File: LeetCode/Study Plan/Practice Problems/Medium/Number Of Zero-Filled Subarrays/Go/number_of_zero_filled_subarrays.go
package main

import "fmt"

func number_of_zero_filled_subarrays(nums []int) int64 {
	num := 0
	res := int64(0)
	for i := 0; i < len(nums); i++ {
		if nums[i] == 0 {
			num++
		} else {
			res += int64(num * (num + 1) / 2)
			num = 0
		}
	}
	if num > 0 {
		res += int64(num * (num + 1) / 2)
	}
	return res
}

func main() {
	fmt.Println(number_of_zero_filled_subarrays([]int{1, 3, 0, 0, 2, 0, 0, 4}))
}

File: LeetCode/Study Plan/Practice Problems/Medium/Number Of Zero-Filled Subarrays/Java/numberOfZeroFilledSubarrays.java
class Solution {
    public long zeroFilledSubarray(int[] nums) {
        int size = nums.length;
        long res = 0;

        for (int i=0; i<size;) {
            if (nums[i] == 0) {
                int k = i;
                long zeros = 0;
                while (k < size && nums[k] == 0) {
                    zeros++;
                    k++;
                }
                res += zeros*(zeros+1)/2;
                i = k;
            } else {
                i++;
            }
        }
        return res;
    }
}
